packages/atomic/src/components/facets/atomic-category-facet/atomic-category-facet.tsx

```diff
@@ -18,34 +20,82 @@ import {
   BaseFacetState,
 } from '../base-facet/base-facet';
 
+import RightArrow from 'coveo-styleguide/resources/icons/svg/arrow-right-rounded.svg';
+import LeftArrow from 'coveo-styleguide/resources/icons/svg/arrow-left-rounded.svg';
+import {
+  FacetSearch,
+  FacetSearchController,
+  FacetSearchState,
+} from '../facet-search/facet-search';
+
+/**
+ * A hierarchical category facet component. It is displayed as a facet in desktop browsers and as
+ * a button which opens a facet modal in mobile browsers.
+ * @part facet - The wrapping div for the entire facet
+ * @part facet-values - The list of facet values (children)
+ * @part facet-value - A single facet value
+ * @part close-button - The button to close the facet when displayed modally (mobile only)
+ * @part reset-button - The button that resets the actively selected facet values
+ * @part show-more - The show more results button
+ * @part show-less - The show less button
+ */
 @Component({
   tag: 'atomic-category-facet',
   styleUrl: 'atomic-category-facet.pcss',
   shadow: true,
 })
 export class AtomicCategoryFacet
-  implements InitializableComponent, BaseFacetState {
+  implements InitializableComponent, FacetSearchState, BaseFacetState {
   @InitializeBindings() public bindings!: Bindings;
-  private facet!: CategoryFacet;
+  public facet!: CategoryFacet;
 
   @BindStateToController('facet', {subscribeOnConnectedCallback: true})
   @State()
-  private facetState!: CategoryFacetState;
+  public facetState!: CategoryFacetState;
   @State() public error!: Error;
 
+  private facetSearch!: FacetSearch;
+
+  @BindStateToI18n()
+  @State()
+  public strings: I18nState = {
+    clear: () => this.bindings.i18n.t('clear'),
+    searchBox: () => this.bindings.i18n.t('facetSearch'),
+    querySuggestionList: () => this.bindings.i18n.t('querySuggestionList'),
+    showMore: () => this.bindings.i18n.t('showMore'),
+    showLess: () => this.bindings.i18n.t('showLess'),
+  };
+
   @State() public isExpanded = false;
-  @Prop({mutable: true, reflect: true}) public facetId = '';
+  @State() public facetSearchQuery = '';
+  @State() public showFacetSearchResults = false;
+
+  /**
+   * Specifies the index field whose values the facet should use
+   */
   @Prop() public field = '';
+  /**
+   * The displayed label for the facet
```

Comment:
I've realized that we might be missing something in atomic to support "localized component options".

What this means is that, for the `label` prop, let's say someone set `label="Category"` in their HTML markup. 
Then, they change the language of the interface to `fr`. 

It would be arguably be better if at the same time the label for the facet adjusted itself to `Catégorie` (assuming that the localization for category string exists/is available).

And not require them to... handle that themselves somehow (not sure how they would manage it).

@ThibodeauJF is this something we could support right now with the localization feature ?

Reply:
@olamothe It's totally doable with the current localization solution, there's a current example for this using the caption for the sort criteria.

In `atomic-sort-dropdown.tsx` line 74 there's an example leveraging the strings state

```tsx
this.options = sortCriterionElements.map(({criteria, caption}) => {
      this.strings[caption] = () => this.bindings.i18n.t(caption);
// ...
```

For the facet label we could do: `this.strings[this.label] = () => this.bindings.i18n.t(this.label);`
And the label would only be the "key" not the string value itself (if translation is needed)
A customer could easily add they own translation after.

As for this PR, im seeing this story https://coveord.atlassian.net/browse/KIT-468 so it might be done later @nrawji ?

Reply:
Yeah the finalization of localization is its own separate task.